src/pyj/read_book/flow_mode.pyj

```diff
@@ -114,15 +119,19 @@ def flow_onwheel(evt):
 
 @check_for_scroll_end
 def goto_boundary(dir):
-    scroll_viewport.scroll_to(window.pageXOffset, 0 if dir is DIRECTION.Up else document_height())
+    scroll_viewport.scroll_to(scroll_viewport.x(), 0 if dir is DIRECTION.Up else document_height())
     get_boss().report_human_scroll()
 
 
 @check_for_scroll_end_and_report
-def scroll_by_page(direction):
+def scroll_by_page(direction, flip_if_rtl_page_progression):
```

Comment:
The functions that call this one know whether or not the action should be reversed in RTL, so I added a flag here.